x-pack/legacy/plugins/uptime/server/rest_api/source_settings/get_source_settings.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { UMServerLibs } from '../../lib/lib';
+import { UMRestApiRouteFactory } from '../types';
+
+export const createGetSourceSettingsRoute: UMRestApiRouteFactory = (libs: UMServerLibs) => ({
+  method: 'GET',
+  path: '/api/uptime/source_settings',
+  validate: false,
+  options: {
+    tags: ['access:uptime'],
+  },
+  handler: async ({ savedObjectsClient: client }, _context, _request, response): Promise<any> => {
+    try {
+      return response.ok({
+        ...await libs.savedObjects.getUptimeSourceSettings(client, undefined)
+      });
+    } catch (e) {
+      return response.internalError({body: {message: e}});
```

Comment:
error details leak to an end user. an example from your Slack message : 
```
"Unsupported saved object type: 'uptime-ui-source': Bad Request"
```

Reply:
OK, will tighten that up.